### PR TITLE
Use new manifest.nextflowVersion instead of params.minNextflowVersion

### PR DESCRIPTION
**Pull request by ewels**
Added support for the new `manifest.nextflowVersion` variable type, see https://github.com/nextflow-io/nextflow/issues/752

This is a more complete version of PR https://github.com/nf-core/cookiecutter/pull/55, updating the linting tools as well as the cookiecutter template.

Note that all pipelines need to be updated to use this new syntax for their tests to pass.

### CODE REVIEW

**Comment by MaxUlysse** [comment]
We should also add the documentation about this feature in the nextflow docs...

**Comment by MaxUlysse** [comment]
Done.
https://github.com/nextflow-io/nextflow/pull/827

File: tests/lint_examples/failing_example/nextflow.config [hunk 1]
```diff
@@ -1,4 +1,5 @@
 manifest.homePage = 'http://nf-co.re/pipelines'
+manifest.nextflowVersion = '0.30.1'
```
[the remark this answers is not in the document]
Comment (apeltzer) [reply tests/lint_examples/failing_example/nextflow.config]: Yes it should be!

[the remark this answers is not in the document]
Comment (MaxUlysse) [reply tests/lint_examples/failing_example/nextflow.config]: Yes, but wasn't it supposed to be a failing example?

**Pull request merged**